app/models/question_type.rb

```diff
@@ -3,79 +3,97 @@
 class QuestionType
   AVAILABLE_PROPERTIES = %w[printable smsable textual headerable defaultable numeric
                             multimedia temporal has_options has_timezone refable reportable].freeze
-  attr_reader :name, :odk_name, :properties
+  attr_reader :name, :odk_name, :odata_type, :properties
 
   @attributes = [{
     name: "text",
     odk_name: "string",
+    odata_type: "string",
```

Comment:
I can see why you put these in here but it's kind of bad coupling I think and the `odk_name` ones actually need to be refactored out of here for the same reason. The better way to do it would be to either have a simple hash somewhere that does the conversion, or if you want something fancier, a decorator/wrapper class that wraps QuestionType and adds the `odata_type` method.